test/he/test_msg_handlers.c

```diff
@@ -169,9 +169,31 @@ void test_msg_handler_ping_both_null(void) {
   TEST_ASSERT_EQUAL(HE_ERR_NULL_POINTER, ret);
 }
 
+void test_msg_handler_ping_invalid_message(void) {
+  conn->state = HE_STATE_ONLINE;
+
+  ret = he_handle_msg_ping(conn, empty_data, 1);
+
+  TEST_ASSERT_EQUAL(HE_ERR_PACKET_TOO_SMALL, ret);
+}
+
 void test_msg_handler_pong(void) {
+  conn->ping_pending_id = 42;
+
   he_internal_generate_event_Expect(conn, HE_EVENT_PONG);
-  ret = he_handle_msg_pong(conn, empty_data, 0);
+
+  he_msg_pong_t *pong = (he_msg_pong_t *)empty_data;
+  pong->id = htons(42);
+  ret = he_handle_msg_pong(conn, pong, sizeof(he_msg_pong_t));
+  TEST_ASSERT_EQUAL(HE_SUCCESS, ret);
+}
+
```

Comment:
Is it possible to add a test that is a round-trip set of a ping generation and pong handling, to guard against future bugs in the Ping / Pong ID matching logic?  For example, if a future commit accidentally saves the `conn->ping_pending_id` in network order instead of host order, then we'll have a bug that the current tests won't catch.

Reply:
It's a bit tricky to add the round-trip test at the unit test level. But I just added tests in `test_send_keepalive_connected` to ensure:

1. ping->id is encoded with network byte order
2. conn->ping_pending_id is set correctly when ping message is sent
3. conn->ping_next_id is incremented